### PR TITLE
raise error for wrong input in ignore_variables

### PR DESCRIPTION
**Pull request by systemshift**
### Pull Request Description
As mentioned in #818, ignore_variables does not raise error when given a wrong input.
Currently it checks if the input is `{str: list}` only, would like to know if its needed to check the values of the list to be string or not.

-----
*After creating the pull request: in order to pass the **changelog_updated** check you will need to update the "Future Release" section of* `docs/source/changelog.rst` *to include this pull request.*

### CODE REVIEW

File: featuretools/synthesis/deep_feature_synthesis.py [hunk 1]
```diff
@@ -165,6 +165,9 @@ def __init__(self,
 
         self.ignore_variables = defaultdict(set)
         if ignore_variables is not None:
+            # check if ignore_variables is not {str: list}
+            if not all(isinstance(i, str) for i in ignore_variables.keys()) or not all(isinstance(i, list) for i in ignore_variables.values()):
+                raise TypeError('ignore_variables should be dict[str -> list[str]]')
```
Comment (rwedge) [inline featuretools/synthesis/deep_feature_synthesis.py]: Could you add a test case that triggers this error?

File: featuretools/tests/synthesis/test_deep_feature_synthesis.py [hunk 2]
```diff
@@ -162,6 +162,18 @@ def test_ignores_variables(es):
         assert 'value' not in variables
 
 
+@pytest.fixture
```
Comment (rwedge) [inline featuretools/tests/synthesis/test_deep_feature_synthesis.py]: This shouldn't be a pytest fixture.  Fixtures are functions that get used by the test cases.  For example, this test case uses the `es` fixture, which returns an entityset.

**Comment by systemshift** [comment]
Now it would be a good time to know if I should add the third check to see if the list is made of exclusively strings, and not say, ints or floats.

**Comment by rwedge** [comment]
That would be good to check as well.

I think only the new "fixes" changelog entry is necessary.     The "testing changes" section is for PRs that only change the tests.

**Comment by rwedge** [comment]
we should also test for the case where not all keys are string type

**Pull request merged**